doc/run-cloud-build-locally.md

```diff
@@ -0,0 +1,197 @@
+c13ad0618e65   hapiproject/hapi:latest   "java --class-path /…"   2 minutes ago        Up 2 minutes        0.0.0.0:9001->8080/tcp   sink-server-search
+1f301480d87f   hapiproject/hapi:latest   "java --class-path /…"   15 minutes ago       Up 15 minutes       0.0.0.0:8091->8080/tcp   hapi-server
+815bc05a6005   postgres                  "docker-entrypoint.s…"   15 minutes ago       Up 15 minutes       0.0.0.0:5432->5432/tcp   hapi-fhir-db
+```
+
+### 2. Compile the Java modules
+
+```bash
+mvn --no-transfer-progress -e -T 2C install -Dlicense.skip=true -Dspotless.apply.skip=true
+```
+
+### 3. Upload sample synthetic data to HAPI stored in
+
+[sample_data](https://github.com/google/fhir-data-pipes/blob/master/synthea-hiv/sample_data)
+to the FHIR server that you brought up using the
+[Synthea data uploader](https://github.com/google/fhir-data-pipes/blob/master/synthea-hiv/README.md#Uploader).
+
+The uploader requires that you install the `uploader` module requirements first.
+You can do this by running:
+
+```bash
+pip3 install -r ./synthea-hiv/uploader/requirements.txt
+```
+
+Please note, it is a good idea to first create a Python `virtualenv` before
+running the above command to avoid conflicts with other Python packages you may
+have installed globally. You can do this by running:
+
+```bash
+$ virtualenv -p python3.8 venv
+$ . ./venv/bin/activate
+```
+
+If you had this already set up, just activate your existing virtualenv.
+
+```bash
+$ workon venv
+```
+
+Then, you can run the uploader script to upload the synthetic data to the FHIR
+server.
+
+For example, to upload to the HAPI FHIR server brought up in the previous step,
+run:
+
+```bash
+python3 ./synthea-hiv/uploader/main.py HAPI http://localhost:9001/fhir \
+--input_dir ./synthea-hiv/sample_data --cores 8
+```
+
+Depending on your machine, using too many cores may slow down your machine or
+cause JDBC connection pool errors with the HAPI FHIR server. Reducing the number
+of cores using the `--cores` flag should help at the cost of increasing the time
+to upload the data.
+
+### 4. Build the e2e controller/spark image
+
+```bash
+docker build -t ${_REPOSITORY}/e2e-tests/controller-spark:${_TAG} -f e2e-tests/controller-spark/Dockerfile .
+```
+
+### 5. Build and run the batch pipeline
+
+Use the Maven-built image (or rebuild with `pipelines/batch/docker` if you
+change it):
+
+```bash
+cd pipelines/batch
+docker build -t ${_REPOSITORY}/batch-pipeline:${_TAG} .
+cd ../..
+```
+
+Run the FHIR-search job against HAPI:
+
+```bash
+docker run --rm \
+  -e FHIR_SERVER_URL=http://localhost:9001/fhir \
+  -e PARQUET_PATH=/workspace/e2e-tests/FHIR_SEARCH_HAPI \
+  -e OUTPUT_PARQUET_VIEW_PATH=/workspace/e2e-tests/FHIR_SEARCH_HAPI/VIEWS_TIMESTAMP_1 \
+  -e SINK_PATH=http://localhost:9001/fhir \
+  -e SINK_USERNAME=hapi -e SINK_PASSWORD=hapi \
+  -v $(pwd):/workspace \
```

Comment:
Consider adding a note about output directories under this path that might be overwritten.

Reply:
I think in this case, it is `PARQUET_PATH` too, see [here](https://github.com/google/fhir-data-pipes/blob/547612dea431ebec8800fb99f9674150e06dce57/pipelines/batch/Dockerfile#L59).